### PR TITLE
Keyboard Searchable Choice Sets

### LINKED ISSUE #298: Implement keyboard support
Currently keyboards are not supported in searchable choice sets or by themselves.

### PR DESCRIPTION
**Pull request by JackLivio**
Fixes #298 (Partial implementation)

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Core Tests
- Send multiple choice sets with images
- Trigger perform interaction with ICON_WITH_SEARCH and LIST_WITH_SEARCH

Core version / branch / commit hash / module tested against: develop
Proxy+Test App name / version / branch / commit hash / module tested against: rpcb-js

### Summary
Introduces a performInteraction view into the keyboard view to allow for searchable choice sets.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/css/components/_keyboard.scss [hunk 1]
```diff
@@ -4,7 +4,29 @@
     display: flex;
     justify-content: center;
     flex-direction: column;
-    padding: 20px;    
+    padding: 20px;
+    padding-top: 0px;
+}
+
+.keyboard .vscrollmenu {
```
Comment (ShobhitAd) [inline src/css/components/_keyboard.scss:11]: Might have to limit the height of the vscrollitem primary image as well
Screenshot from 2022-02-02 14-47-46

File: src/css/components/_keyboard.scss [hunk 2]
```diff
@@ -4,7 +4,29 @@
...
+    height: 40%;
+}
+
+.keyboard .hscrollmenu {
+    padding-top: 10px;
+    height: 40%;
+    display: flex;
+    justify-content: space-between;
```
Comment (ShobhitAd) [inline src/css/components/_keyboard.scss:19]: Might have to limit the height of the hscrollitem primary image as well
Screenshot from 2022-02-02 15-46-07

File: src/js/Keyboard.js [hunk 3]
```diff
@@ -234,20 +227,41 @@ class Keyboard extends Component {
 
     var interactionText = app && app.interactionText && app.interactionText.fieldText ? 
       app.interactionText.fieldText : "Tap on the virtual keyboard to start";
+
+    var parsedInput = this.state.input.replaceAll("\t", "        ");
+    var calculatedInputSize = parsedInput ? parsedInput.length : interactionText.length;
+
+    var inputClassName = "div-input"
+    if (parsedInput.length && (this.maskedInput || this.state.userMaskedInput)) {
+      inputClassName += " text-security"
```
Comment (ShobhitAd) [inline src/js/Keyboard.js:236]: Shouldn't the input be displayed to the user while using `LIST_WITH_SEARCH`/`ICON_WITH_SEARCH`. Any reason as to why we'd want to obfuscate the search string.

**Review by ShobhitAd — CHANGES_REQUESTED** [review]

File: src/js/Keyboard.js (hunk 3 above)
Comment (JackLivio) [reply src/js/Keyboard.js:236]: So these changes were more for fixing the masked input when i switched from a input html field to a plain ol div. This pr is based of those changes. That brings up a good question on why a dev would ever want this, but thats kinda how the keyboard properties are supposed to work i think

File: src/css/components/_keyboard.scss (hunk 2 above)
Comment (ShobhitAd) [reply src/css/components/_keyboard.scss:19]: The primary image looks good with the changes but might have to apply similar constraints to the secondary image
Screenshot from 2022-02-07 09-56-17

**Review by ShobhitAd — CHANGES_REQUESTED** [review]

**Review by ShobhitAd — APPROVED** [review]

**Pull request merged**